m3/src/test/java/com/uber/m3/tally/m3/M3ReporterTest.java

```diff
@@ -49,6 +51,9 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+// TODO add tests to validate proper shutdown
```

Comment:
do you want to add those tests in this PR/following PR?

Reply:
Please feel free to pick it up

Reply:
sg